Require sudo mode on the provisioning API

nickvergessen opened the pull request:
Ref #2487 

@LukasReschke @rullzer

nickvergessen committed:
Require sudo mode on the provisioning API

Signed-off-by: Joas Schilling <coding@schilljs.com>

nickvergessen commented on apps/provisioning_api/lib/Controller/GroupsController.php:
```diff
@@ -128,7 +128,7 @@ public function getGroup($groupId) {
 	/**
 	 * creates a new group
 	 *
-	 * @NoAdminRequired
```
@rullzer you added that by accident in https://github.com/nextcloud/server/commit/432e7c93c6dad564abbaec1e3d374f73653d7ba6

This allows all users to create groups Oo

rullzer replied:
Ah mmm my bad

LukasReschke replied:
@nickvergessen ❤️

LukasReschke commented:
>error_outlineERROR: 500 Internal Server Error: write /docker/aufs/layers/8d99f89faa9f651a88d3d69ce07cb8917cba2652143e34ce37d0bf961746fe66-init: no space left on device

Let's retrigger the tests.

cc @MorrisJobke

MorrisJobke commented:
> Let's retrigger the tests.

:(

MorrisJobke commented:
Provisioning API integration tests say 💥

MorrisJobke commented:
> Provisioning API integration tests say 💥

And I also think that this will cause a lot of broken scripts - we should add this to the release notes to avoid trouble here and guiding people towards the app tokens ;)

rullzer commented:
Mmmm why do they 💥 ? Because we use basic auth on OCS... or does do the intergration tests use dark cookie magic?

LukasReschke commented:
> And I also think that this will cause a lot of broken scripts - we should add this to the release notes to avoid trouble here and guiding people towards the app tokens ;)

Scripts should use Basic Auth as specified in the OCS specs. :)

MorrisJobke commented:
@LukasReschke Ah that means HTTP Auth is fine for this. I thought it needs to be an app specific password to not catch the password confirmation ... got it :)

nickvergessen commented:
Will have a look at the tests

nickvergessen commented:
Seems like we dont use basic auth, but changing it doesn't work for me either. Also my xdebug fails to debug those tests, so maybe someone else can have a look

MorrisJobke commented:
> Seems like we dont use basic auth, but changing it doesn't work for me either. Also my xdebug fails to debug those tests, so maybe someone else can have a look

cc @rullzer ?

LukasReschke commented:
Will take a look.

rullzer committed:
Set last-login-check on basic auth

Else the last-login-check fails hard because the session value is not
set and thus defaults to 0.

* Started with tests

Signed-off-by: Roeland Jago Douma <roeland@famdouma.nl>

rullzer commented:
Fixed. @LukasReschke please verify my change

LukasReschke commented:
Nice, even with tests 🚀

LukasReschke commented:
LGTM

MorrisJobke commented:
:+1:
